samples: drivers/net/: apps: Resolved Kconfig dependencies

SavinayDharmappa opened the pull request:
patch add missing dependency CONFIG_FLASH and removes CONFIG_GPIO
as it is selected by default. This patch is a fix for jira ZEP2071

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

nashif reviewed (changes requested):
That does not seem right. This sample does not require FLASH and does not depend on it. the problem is elsewhere, basically something is selecting GPIO where it should just depend on it.

SavinayDharmappa committed:
samples: drivers/net: apps: Resolve Kconfig dependency

as config SPI_CS_GPIO was selecting GPIO instead it
should just depend on it. This patch is a fix for
ZEP-2071 jira.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>
